sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/EncodingPerFileTest.java

```diff
@@ -51,40 +50,44 @@ public void should_treat_as_match_when_roslyn_entry_missing_for_file() throws IO
     Charset roslynCharset = StandardCharsets.UTF_8;
     Charset sqCharset = StandardCharsets.UTF_16;
 
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, roslynCharset);
-    assertEncodingMatch(encodingPerUri, URI.create("dummy"), sqCharset, true);
+    assertEncodingMatch(roslynCharset, URI.create("dummy"), sqCharset, true);
   }
 
   @Test
   public void should_treat_as_mismatch_when_roslyn_encoding_missing() throws IOException {
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, null);
-    assertEncodingMatch(encodingPerUri, fileUri, null, false);
+    assertEncodingMatch(null, fileUri, null, false);
   }
 
   @Test
   public void should_treat_as_mismatch_when_roslyn_utf8_and_sq_utf16() throws IOException {
     Charset roslynCharset = StandardCharsets.UTF_8;
     Charset sqCharset = StandardCharsets.UTF_16;
 
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, roslynCharset);
-    assertEncodingMatch(encodingPerUri, fileUri, sqCharset, false);
+    assertEncodingMatch(roslynCharset, fileUri, sqCharset, false);
   }
 
   @Test
   public void should_treat_as_match_when_roslyn_utf16_and_sq_utf16le() throws IOException {
     Charset roslynCharset = StandardCharsets.UTF_16;
     Charset sqCharset = StandardCharsets.UTF_16LE;
 
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, roslynCharset);
-    assertEncodingMatch(encodingPerUri, fileUri, sqCharset, true);
+    assertEncodingMatch(roslynCharset, fileUri, sqCharset, true);
   }
 
-  private void assertEncodingMatch(Map<URI, Charset> encodingPerUri, URI fileUri, Charset sqCharset, boolean result) throws IOException {
+  @Test
+  public void encoding_per_file_is_not_case_sensitive() throws IOException {
+    Charset sqCharset = StandardCharsets.UTF_16;
+
+    assertEncodingMatch(StandardCharsets.UTF_8, URI.create(fileUri.toString().toUpperCase()), sqCharset, false);
+    assertEncodingMatch(StandardCharsets.UTF_8, URI.create(fileUri.toString().toLowerCase()), sqCharset, false);
+    assertEncodingMatch(StandardCharsets.UTF_16, URI.create(fileUri.toString().toUpperCase()), sqCharset, true);
+    assertEncodingMatch(StandardCharsets.UTF_16, URI.create(fileUri.toString().toLowerCase()), sqCharset, true);
+  }
+
+  private void assertEncodingMatch(Charset roslynCharset, URI fileUri, Charset sqCharset, boolean result) throws IOException {
     AbstractGlobalProtobufFileProcessor processor = mock(AbstractGlobalProtobufFileProcessor.class);
+    TreeMap<String, Charset> encodingPerUri = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
I am confused by the fact that `encoding_per_file_is_not_case_sensitive` uses, via `assertEncodingMatch`, a `TreeMap`

if you change the `TreeMap` with a `HashMap` here on line 89, will it the case sensitivity test fail?

because if it does, it seems the unit test setup makes it case insensitive rather than the system under test

Reply:
Yes, the test setup takes care of this. The case-insensivity of the TreeMap itself is already tested by `AbstractGlobalProtobufFileProcessorTest.is_not_case_sensitive`. Here we just need to mock the same behavior.

This test has currently value as a safety net in case that non-trivial `encodingMatch()` is reimplemented in some way that will break this. So I'd like to keep this test and don't delete it.

Reply:
I find it weird having a test which is testing a behavior enforced by the underlying mocks.

I'm ok with keeping the test, but please add a comment to mention this fact. Someone seeing this later on may be tempted to remove the test (or may be confused).